Marlin/src/libs/autoreport.h

```diff
@@ -25,23 +25,25 @@
 
 template <typename Helper>
 struct AutoReporter {
-  millis_t next_report_ms;
+  millis_t last_report_ms;
```

Comment:
We prefer to use "next" over "last" because it produces more efficient code.

Reply:
> We prefer to use "next" over "last" because it produces more efficient code.

Yep, it seems to produce one 32bit comparison in the loop. And then one time 32bit addition when intervalt completes.
Versus 32bit subtraction and 32bit comparison, in the loop. And then one 32bit copy when interval completes.

Downside is that the more efficient version causes causes issues when program is running long enough (counting from the initial start of the program). 

The efficiency difference seem not to be big here. It would need analysing by observing disassembly listing. I do not recall how big the differece were, though I have done the analysis.

It would be nice to have the disclaimer in the coding standards that the "next" style only be used for intermittent items, because there the pros outweigh the cons. That when using the more efficient "next" version, in the constantly running interval generators it would surely crash when millis() overflow moment reaches.